openapi_metadata.go

```diff
@@ -0,0 +1,449 @@
+package fuego
+
+import (
+	"encoding/xml"
+	"fmt"
+	"log/slog"
+	"reflect"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+const (
+	metadataParserExample     = "exampleParser"
+	metadataParserValidation  = "validationParser"
+	metadataParserDescription = "descriptionParser"
+	metadataParserXML         = "XMLParser"
+	metadataParserJSON        = "JSONParser"
+)
+
+var DefaultParsers = []MetadataParserEntry{
+	{Name: metadataParserExample, Parser: MetadataParserExample},
+	{Name: metadataParserValidation, Parser: MetadataParserValidation},
+	{Name: metadataParserDescription, Parser: MetadataParserDescription},
+	{Name: metadataParserXML, Parser: MetadataParserXML},
+	{Name: metadataParserJSON, Parser: MetadataParserJSON},
+}
+
+type MetadataParserParams struct {
+	// Reflective information of a struct field
+	Field reflect.StructField
+	// Name of the field
+	FieldName string
+	// OpenAPI schema of the field
+	Property *openapi3.Schema
+	// Reference to the OpenAPI schema
+	SchemaRef *openapi3.SchemaRef
+	// Additional metadata as key-value pairs
+	Additional map[string]interface{}
+}
+
+type MetadataParserFunction func(params MetadataParserParams)
+
+type MetadataParserEntry struct {
+	Name   string
+	Parser MetadataParserFunction
+}
+
+type MetadataParsers struct {
+	registeredParsers []MetadataParserEntry
+	parserLock        sync.Mutex
+	registeredNames   map[string]bool
+}
+
+// NewMetadataParsers returns a new MetadataParsers instance with empty registeredParsers
+// slice and registeredNames map.
+func NewMetadataParsers() *MetadataParsers {
+	return &MetadataParsers{
+		registeredParsers: []MetadataParserEntry{},
+		registeredNames:   make(map[string]bool),
+	}
+}
+
+// GetRegisteredParsers returns a slice of all currently registered metadata parsers.
+// This method ensures thread-safe access to the registeredParsers slice by using a mutex lock.
+func (mp *MetadataParsers) GetRegisteredParsers() []MetadataParserEntry {
+	mp.parserLock.Lock()
+	defer mp.parserLock.Unlock()
+
+	return mp.registeredParsers
+}
+
+// Reset clears all registered metadata parsers from the slice and resets the
+// registeredNames map as empty. This method ensures thread-safe access to the
+// registeredParsers slice and registeredNames map by using a mutex lock.
+func (mp *MetadataParsers) Reset() {
+	mp.parserLock.Lock()
+	defer mp.parserLock.Unlock()
+
+	mp.registeredParsers = []MetadataParserEntry{}
+	mp.registeredNames = make(map[string]bool)
+}
+
+// InitializeMetadataParsers initializes the metadata parsers with the given customParsers.
```

Comment:
Please update this accordingly

```suggestion
// Initialize initializes the metadata parsers with the given customParsers.
```